types/swap.d.ts

```diff
@@ -0,0 +1,58 @@
+import { Placeholder } from './util/tools';
+
+// swap(indexA)
+export function swap(indexA: number): {
+  // swap(indexA)(indexB)(list)
+  (indexB: number): <T>(list: readonly T[]) => T[];
+  // swap(indexA)(__, list)(indexB)
+  <T>(__: Placeholder, list: readonly T[]): (indexB: number) => T[];
+  // swap(indexA)(indexB, list)
+  <T>(indexB: number, list: readonly T[]): T[];
+};
+
+// swap(__, indexB)
+export function swap(
+  __: Placeholder,
+  indexB: number
+): {
+  // swap(__, indexB)(indexA)(list)
+  (indexA: number): <T>(list: readonly T[]) => T[];
```

Comment:
I'm curious about the `readonly` keyword here.  Is this just preventing the definition of `swap` from mutating the list argument?  I guess the implementation of `swap` is already written in the other repo and these definitions do not restrict that implementation.  

Does `readonly` do anything else that restricts the consumer or type inference in any way?  It probably can't hurt to have these here, but just wanted to see if I was missing something else.

Reply:
ahh, that makes sense.   It does help the consumer in that this function now accepts `readonly` arrays.  Thanks for the explanation!

Reply:
[playground example](https://tsplay.dev/WzxjeN)